pkg/scheduler/eventhandlers.go

```diff
@@ -176,11 +176,15 @@ func (sched *Scheduler) addPodToSchedulingQueue(obj interface{}) {
 }
 
 func (sched *Scheduler) updatePodInSchedulingQueue(oldObj, newObj interface{}) {
-	pod := newObj.(*v1.Pod)
-	if sched.skipPodUpdate(pod) {
+	new := newObj.(*v1.Pod)
+	if sched.skipPodUpdate(new) {
 		return
 	}
-	if err := sched.SchedulingQueue.Update(oldObj.(*v1.Pod), pod); err != nil {
+	old := oldObj.(*v1.Pod)
+	if old.ResourceVersion == new.ResourceVersion {
```

Comment:
Why would we get this, and why wouldn't `skipPodUpdate` skip it?

Reply:
Enforced sync would trigger an update event carrying two identical objects.

Reply:
how come this is not enough to cover that case? https://github.com/kubernetes/kubernetes/blob/d1c296431e0ff2363131707054c4c75ad59cd2c0/pkg/scheduler/eventhandlers.go#L334

There is probably some other field we should be excluding, instead of comparing the resource version.

Reply:
`skipPodUpdate()` deals with assumed pod only, right? It's very likely the update event carries a non-assumed pod. Let me put some log to verify it.